scripts/clean-data.py

```diff
@@ -132,11 +166,28 @@ def clean_fqn_test():
     'fastparse.Api.P(scala.Functi_on0,sourcecode.Name)fastparse.core.Parser'
     '''
 
+clean_signature = compose(
+    remove_arrow,
+    remove_hashtags,
+    remove_L_notation,
+    remove_trailing_dot,
+    remove_leading_root
+)
+def clean_signature_test():
+    '''
+    An empty function to trigger doctest.
+    clean_signature agglomerates various cleaning functions to clean the resolved type name from semanticdb.
+
+    >>> clean_signature('[T, R] => (x: _root_.scala.Tuple2#[_root_.scala.concurrent.Future#[T], _root_.scala.concurrent.Future#[R]])(implicit executionContext: _root_.scala.concurrent.ExecutionContext#): _empty_.PipeableFuture#[T, R]')
+    '[T, R](x: scala.Tuple2[scala.concurrent.Future[T], scala.concurrent.Future[R]])(implicit executionContext: scala.concurrent.ExecutionContext): PipeableFuture[T, R]'
+    '''
+
 def clean_param_row(row):
     row["fqn"] = clean_fqn(row["fqn"])
     row["fqfn"] = extract_function_name(row["fqn"])
```

Comment:
is the function name any different from name?

Reply:
yes, the function name is given in `fully.qualified.form`, and the name is just what comes after the last dot and before the first '('

Reply:
so what is `fqn`?

Reply:
If we eventually transition to a parameter table, we can probably do something more useful with `fqfn`

Reply:
`fqn` is the complete symbol `path.to.fun(param.one.type,param.two.type)return.val.type`.

Reply:
OK, here is confusion. I though that it is really just fully qualified name which together with signature makes an id.

Reply:
Wouldn't that be enough?